src/server.rs

```diff
@@ -297,7 +320,7 @@ impl Stream for CoAPServer {
     type Item = Result<Message, io::Error>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        if let Poll::Ready(Some((p, a))) = self.receiver.poll_next_unpin(cx) {
+        if let Poll::Ready(Some((p, a))) = self.receiver.poll_recv(cx) {
```

Comment:
So I narrowed down the problematic code to this line. This was a silly mistake done by me, but I'm not able to understand how to move forward. Essentially the `CoAPServer.receiver` is currently of the type `UnboundedReceiver`, but needs to be wrapped by `UnboundedReceiverStream`. I tried changing the types, but I was running into bizarre issues. 

@Covertness I'm sure you're quite busy, but is it possible for you to take a look at this? Hopefully it won't take much time